flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/model/KinesisStreamShard.java

```diff
@@ -61,19 +68,23 @@ public String getShardId() {
 	}
 
 	public String getStartingSequenceNumber() {
-		return shard.getSequenceNumberRange().getStartingSequenceNumber();
+		SequenceNumberRange sequenceNumberRange = shard.getSequenceNumberRange();
+		return (sequenceNumberRange == null) ? null : sequenceNumberRange.getStartingSequenceNumber();
```

Comment:
I added these null checks because in the tests, fake `KinesisStreamShards` may be created without giving them a SequenceNumberRange / HashNumberRange etc. if it wasn't relevant to the test. Perhaps I should revert these null check changes, and always give all values for the fake `KinesisStreamShard`s?